src/network/networkd-link.c

```diff
@@ -2501,13 +2501,65 @@ static bool link_address_is_dynamic(Link *link, Address *address) {
         return false;
 }
 
+static int link_enumerate_ipv6_tentative_addresses(Link *link) {
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL, *reply = NULL;
+        sd_netlink_message *addr;
+        int r;
+
+        assert(link);
+        assert(link->manager);
+        assert(link->manager->rtnl);
+
+        r = sd_rtnl_message_new_addr(link->manager->rtnl, &req, RTM_GETADDR, 0, AF_INET6);
```

Comment:
I think setting `ifindex` here is more efficient.

Reply:
I tried that, but the kernel still replies will addresses for all interfaces...possibly because any use of RTM_GETADDR automatically sets NLM_F_DUMP?  I could check again and/or look in the kernel.  I also tried using sd_netlink_message_set_flags() to only request IFA_F_TENTATIVE addresses, but that also didn't work - the response included tentative and non-tentative addresses.  I even tried clearing NLM_F_DUMP but the netlink call didn't work then - I think the RTM_GETADDR call requires one (or both) of the NLM_F_DUMP bits to be set (it's just NLM_F_ROOT | NLM_F_MATCH)

Reply:
I *think* it may be possible to have the kernel limit/filter what addresses it replies with, but a quick check of the kernel seems to indicate it will require setting the NETLINK_F_STRICT_CHK flag on the netlink socket:
https://gitlab.freedesktop.org/drm/msm/commit/89d35528d17d25819a755a2b52931e911baebc66

I'm not sure if that would affect other netlink calls, or if it's worth doing for this.  It might be worth doing this later, separately, since it seems like that's the intention of the new kernel socket flag, but as it might affect other parts of systemd it probably shouldn't be done with this patch.

Reply:
OK. Let's do that in a later PR then.